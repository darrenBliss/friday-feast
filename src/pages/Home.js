// @flow
import React, { Component } from 'react'
import DayOfWeek from '../components/DayOfWeek'

class Home extends Component<void> {
  constructor() {
    super()
  }

  render() {
    return (
      <div>
        <h1>Homepage</h1>
        <p>Whose turn is it this friday?</p>
        <DayOfWeek dayIndex={5} />
      </div>
    )
  }
}

export default Home