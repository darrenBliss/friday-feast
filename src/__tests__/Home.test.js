import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import Home from '../pages/Home'

jest.mock('../fire', () => {

  const firebasemock = require('firebase-mock');

  const mockdatabase = new firebasemock.MockFirebase();
  const mockauth = new firebasemock.MockFirebase();
  const mocksdk = new firebasemock.MockFirebaseSdk(path => {
    return path ? mockdatabase.child(path) : mockdatabase;
  }, () => {
    return mockauth;
  });

  const firebase = mocksdk.initializeApp(); // can take a path arg to database url
  // optional - expose the mock
  global.firebase = firebase;

  // return the mock to match your export api
  return firebase;
});

describe('Should render Home page/component', () => {

  it('snapshot should work', () => {
    const wrapper = mount(<Home />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })


})
