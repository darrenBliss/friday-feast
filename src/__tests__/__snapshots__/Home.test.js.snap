// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Home page/component snapshot should work 1`] = `
<Home>
  <styled.div>
    <div
      className="sc-bdVaJa qNUtU"
    >
      <styled.p>
        <p
          className="sc-bwzfXH dlIQYB"
        >
          üóì This week: 
          <DayOfWeek
            dayIndex={5}
          >
            
      Friday
      11
      May
      2018
    
          </DayOfWeek>
        </p>
      </styled.p>
      <styled.div>
        <div
          className="sc-ifAKCX cTeqPF"
        >
          <div>
            <styled.div>
              <div
                className="sc-bZQynM iTFdCi"
              />
            </styled.div>
            <p>
              It is your turn to pick this week.
            </p>
          </div>
          <styled.div
            role="presentation"
          >
            <div
              className="sc-EHOje hytPIO"
              role="presentation"
            >
              üçî
            </div>
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Home>
`;
